src/platform/packages/shared/kbn-esql-ast/src/definitions/utils/autocomplete/expressions/SuggestionBuilder.ts

```diff
@@ -0,0 +1,213 @@
+      addSpaceAfterField,
+      openSuggestions,
+      addComma,
+      promoteToTop,
+      values,
+    });
+
+    this.suggestions.push(...fieldSuggestions);
+    return this;
+  }
+
+  addFunctions(options?: {
+    types?: FunctionParameterType[];
+    ignoredFunctions?: string[];
+    addComma?: boolean;
+    addSpaceAfterFunction?: boolean;
+    openSuggestions?: boolean;
+  }): this {
+    const types = options?.types ?? ['any'];
+    const ignored = options?.ignoredFunctions ?? [];
+    const addSpaceAfterFunction = options?.addSpaceAfterFunction;
+    const openSuggestions = options?.openSuggestions;
+
+    const functionSuggestions = getFunctionsSuggestions({
+      location: this.context.location,
+      types,
+      options: {
+        ignored,
+        addComma: options?.addComma,
+        addSpaceAfterFunction,
+        openSuggestions,
+      },
+      context: this.context.context,
+      callbacks: {
+        hasMinimumLicenseRequired: this.context.callbacks?.hasMinimumLicenseRequired,
+      },
+    });
+
+    this.suggestions.push(...functionSuggestions);
+    return this;
+  }
+
+  addLiterals(options?: {
+    types?: FunctionParameterType[];
+    addComma?: boolean;
+    includeDateLiterals?: boolean;
+    includeCompatibleLiterals?: boolean;
+    advanceCursorAndOpenSuggestions?: boolean;
+  }): this {
+    const types = options?.types ?? ['any'];
+    const includeDateLiterals = options?.includeDateLiterals ?? true;
+    const includeCompatibleLiterals = options?.includeCompatibleLiterals ?? true;
+    const advanceCursorAndOpenSuggestions = options?.advanceCursorAndOpenSuggestions ?? false;
+
+    const literals = getLiteralsSuggestions(types, this.context.location, {
+      includeDateLiterals,
+      includeCompatibleLiterals,
+      addComma: options?.addComma,
+      advanceCursorAndOpenSuggestions,
+    });
+
+    this.suggestions.push(...literals);
+    return this;
+  }
+
+  addOperators(options?: {
+    leftParamType?: FunctionParameterType;
+    allowed?: string[];
+    ignored?: string[];
+    returnTypes?: SupportedDataType[];
+  }): this {
+    const operatorSuggestions = getOperatorSuggestions(
+      {
+        location: this.context.location,
+        leftParamType: options?.leftParamType,
+        allowed: options?.allowed,
+        ignored: options?.ignored,
+        returnTypes: options?.returnTypes,
+      },
+      this.context.callbacks?.hasMinimumLicenseRequired,
+      this.context.context?.activeProduct
+    );
+
+    this.suggestions.push(...operatorSuggestions);
+    return this;
+  }
+
+  /**
+   * Adds comma suggestion based on decision engine rules.
+   */
+  addCommaIfNeeded(commaContext: CommaContext): this {
+    if (shouldSuggestComma(commaContext)) {
+      this.suggestions.push(commaCompleteItem);
+    }
+
+    return this;
+  }
+
+  /**
+   * Handles suggestions for unknown type expressions within function parameters.
+   */
+  async addUnknownTypeSuggestions(): Promise<this> {
```

Comment:
Do we really need to cover this case? It seems like going a little far to make these guesses if the column doesn't exist. Let me know if I missed something.

<img width="804" height="516" alt="Image" src="https://github.com/user-attachments/assets/e339e3a7-ac1a-46aa-8cc2-1c6a686fb0d4" />